node/src/main/java/de/obsidiancloud/node/module/ModuleLoader.java

```diff
@@ -0,0 +1,284 @@
+                if (manifest2.getDependencies().contains(manifest.getName())
+                        || manifest2.getSoftDependencies().contains(manifest.getName())) {
+                    priority = i;
+                    break;
+                }
+            }
+            sorted.add(priority, manifest);
+        }
+        return sorted;
+    }
+
+    /**
+     * Enables a module
+     *
+     * @param manifest The manifest of the module to enable
+     */
+    public static void enableModule(@NotNull ModuleManifest manifest, Logger logger) {
+        try {
+            for (String dependency : manifest.getDependencies()) {
+                if (getModule(dependency) == null) {
+                    logger.warning(
+                            "Module \""
+                                    + manifest.getName()
+                                    + "\" has a dependency to \""
+                                    + dependency
+                                    + "\" which is not loaded");
+                    return;
+                }
+            }
+            logger.info("Enabling module \"" + manifest.getName() + "\"");
+            if (!manifest.isEnabled()) {
+                loadClasses(manifest);
+            }
+            manifest.getInstance().onEnable();
+        } catch (Exception exception) {
+            logger.log(
+                    Level.SEVERE,
+                    "An error occurred while enabling " + manifest.getName(),
+                    exception);
+            disableModule(manifest, logger);
+        }
+    }
+
+    /**
+     * Disables a module
+     *
+     * @param manifest The manifest of the module to disable
+     */
+    public static void disableModule(@NotNull ModuleManifest manifest, @NotNull Logger logger) {
+        try {
+            logger.info("Disabling module \"" + manifest.getName() + "\"");
+            for (ModuleManifest manifest2 : manifests) {
+                if (manifest2.getDependencies().contains(manifest.getName())
+                        && manifest2.isEnabled()) {
+                    disableModule(manifest2, logger);
+                }
+            }
+            if (manifest.isEnabled()) {
+                manifest.getInstance().onDisable();
+            }
+        } catch (Exception exception) {
+            logger.log(
+                    Level.SEVERE,
+                    "An error ocurred while disabling " + manifest.getName(),
+                    exception);
+        }
+        if (manifest.isEnabled()) {
+            unloadClasses(manifest, logger);
+        }
+    }
+
+    /**
+     * Loads all classes from a jar file and initializes the main class
+     *
+     * @param manifest The manifest of the module
+     * @throws IOException If an error occurs while loading the classes
+     */
+    @SuppressWarnings("unchecked")
+    public static void loadClasses(@NotNull ModuleManifest manifest)
+            throws IOException,
```

Comment:
Am besten alle Exceptions mit spezifischen Logs in dieser Methode abfangen anstatt throw